hydpy/models/rconc/rconc_model.py

```diff
@@ -0,0 +1,288 @@
+        >>> derived.dt.update()
+        >>> states.sc = 0.0
+        >>> model.calc_outuh_sc_v1()
+        >>> fluxes.outuh
+        outuh(0.026159)
+        >>> states.sc
+        sc(0.850033, 0.590099, 0.327565, 0.149042, 0.057103)
+        >>> round_(fluxes.outuh + sum(states.sc))
+        2.0
+    """
+
+    CONTROLPARAMETERS = (rconc_control.NmbStorages, rconc_control.RecStep)
+    DERIVEDPARAMETERS = (rconc_derived.DT, rconc_derived.KSC)
+    REQUIREDSEQUENCES = (rconc_fluxes.InUH,)
+    UPDATEDSEQUENCES = (rconc_states.SC,)
+    RESULTSEQUENCES = (rconc_fluxes.OutUH,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model) -> None:
+        con = model.parameters.control.fastaccess
+        der = model.parameters.derived.fastaccess
+        flu = model.sequences.fluxes.fastaccess
+        sta = model.sequences.states.fastaccess
+        if (con.nmbstorages == 0) or modelutils.isinf(der.ksc):
+            flu.outuh = flu.inuh
+        else:
+            flu.outuh = 0.0
+            for _ in range(con.recstep):
+                sta.sc[0] += der.dt * flu.inuh
+                for j in range(con.nmbstorages - 1):
+                    d_q = min(der.dt * der.ksc * sta.sc[j], sta.sc[j])
+                    sta.sc[j] -= d_q
+                    sta.sc[j + 1] += d_q
+                j = con.nmbstorages - 1
+                d_q = min(der.dt * der.ksc * sta.sc[j], sta.sc[j])
+                sta.sc[j] -= d_q
+                flu.outuh += d_q
+
+
+class Set_RConcInput_V1(modeltools.Method):
+    """Set the input for the calculation of the runoff concentration."""
+
+    RESULTSEQUENCES = (rconc_fluxes.InUH,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model, v: float) -> float:
+        flu = model.sequences.fluxes.fastaccess
+        flu.inuh = v
+
+
+class Determine_RConcOutput_V1(modeltools.Method):
+    """Interface method that applies the complete application model by executing all
+    "run methods"."""
+
+    @staticmethod
+    def __call__(model: modeltools.AdHocModel) -> None:
+        model.run()
+
+
+class Get_RConcOutput_V1(modeltools.Method):
+    """Get the previously calculated runoff concentration output.
+
+    Example:
+
+        >>> from hydpy.models.rconc import *
+        >>> parameterstep("1d")
+        >>> simulationstep("12h")
+        >>> tb(3.0)
+        >>> fluxes.outuh = 2.0
+        >>> model.get_rconcoutput_v1()
+        2.0
+    """
+
+    REQUIREDSEQUENCES = (rconc_fluxes.OutUH,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model) -> float:
+        flu = model.sequences.fluxes.fastaccess
+
+        return flu.outuh
+
+
+class Get_WaterBalance_V1(modeltools.Method):
+    """Get the water balance if triangle unit hydrograph is used."""
+
+    REQUIREDSEQUENCES = (rconc_logs.QUH,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model, initial_conditions: ConditionsSubmodel) -> float:
+        return model.sequences.logs.quh - initial_conditions["logs"]["quh"]
```

Comment:
Better implement it similarly to `check_waterbalance` as a "normal" Python method.

Reply:
ok